Sources/web3swift/Transaction/AbstractEnvelope.swift

```diff
@@ -0,0 +1,133 @@
+// Package: web3swift
+// Created by Alex Vlasov.
+// Copyright © 2018 Alex Vlasov. All rights reserved.
+//
+// Support for EIP-2718 by Mark Loit 2022
+
+import Foundation
+import BigInt
+
+/// Enumeration for supported transaction types
+/// .legacy for untyped and type 0 transactions EIP155 and older
+/// .eip2930 for type 1 transactions conforming to EIP2930
+/// .eip1559 for type 2 transactions conforming to EIP1559
+public enum TransactionType: UInt, CustomStringConvertible {
+    case legacy
+    case eip2930
+    case eip1559
+
+    case unknown // catch-all type, always place as last position, also used for range checking
+    // only used in parsing, no TransactionObject will ever return this
+
+    public var description: String {
+        switch self {
+        case .legacy: return "Legacy"
+        case .eip2930: return "EIP-2930"
+        case .eip1559: return "EIP-1559"
+        case .unknown: return "Unknown EIP-2718 Type"
+        }
+    }
+}
+
+/// Encoding selector for transaction transmission/hashing or signing
+public enum EncodeType {
+    case transaction
+    case signature
+}
+
+/// Protocol definition for all transaction envelope types
+///  All envelopes must conform to this protocol to work with EthereumTransaction
+///   each implememtation holds all the type specific data
+///   and implments the type specific encoding/decoding
+public protocol AbstractEnvelope: CustomStringConvertible { // possibly add Codable?
+    /// The type of transaction this envelope represents (.legacy, .eip2930, .eip1559)
```

Comment:
Suggest removing `... (.legacy, .eip2930, .eip1559)`. Cases of `TransactionType` speak for themselves.